File: Java/Assignment9_4.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment9_4.java
// Description : write a program which Accept number from user and count the frequency of 4
// input.      : 2395
// output.     : 0
// Author      : Aniket Devaba Markad                        
// Date        : 1/1/2026
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class  Digit
{

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : CountDigit
// Description   : This function is used to count the frequency of 4
// Author        : Aniket Devaba Markad 
// Date          : 1/1/2026 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    int CountDigit(int iNo)
    {
        int iDigit = 0,iCnt = 0;
        if(iNo < 0)
        {
            iNo = -iNo;
        }
       
        while(iNo != 0)
        {

        iDigit = iNo % 10;
        if(iDigit == 4 )
        {
            iCnt++;
        }
        iNo = iNo/10;
        }
        return iCnt;
    }
   
}
public class Assignment9_4
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("enter the number  ");
        int iValue = sobj.nextInt();

        Digit dobj = new Digit();
        int iRet = dobj.CountDigit(iValue);

        System.out.println("frequency of 4 is :" + iRet);

    }
}

File: CPP/Assignment9_4.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment9_4.cpp
// Description : Write a program which accept number from user and check whether it contains 0 in it or not
// Author      : Aniket Devaba Markad
// Date        : 12/12/2025
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;
#define TRUE 1
#define FALSE 0
typedef int BOOL;

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// 
// Function Name : ChkZero
// Description   : This function is used to check whether it contains 0 in it or not
// Input         : int
// Output        : BOOL
// Author        : Aniket Devaba Markad
// Date          : 12/12/2025
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

BOOL ChkZero(int iNo)
{
    int iDigit = 0;

    if(iNo<0)
    {
        iNo=-iNo;
    }
    while (iNo!=0)
    {
        iDigit=iNo%10;
        if(iDigit==0)
        {
            return TRUE;
        }
        iNo = iNo / 10;
    }
    return FALSE;
}

int main()
{
    int iValue =0;
    BOOL bRet = FALSE;

    cout<<"enter number";
    cin>>iValue;

    bRet= ChkZero(iValue);
    if(bRet == TRUE)
    {
        cout<<"it contains zero";
    }
    else
    {
        cout<<"There is no zero";
    }
    return 0;
}
